File: week-02/day-1/src/DiagonalMatrix.java
import java.util.Arrays;

public class DiagonalMatrix {

  public static void main(String[] args) {
    // - Create (dynamically) a two dimensional array
    //   with the following matrix. Use a loop!
    //
    //   1 0 0 0
    //   0 1 0 0
    //   0 0 1 0
    //   0 0 0 1
    //
    // - Print this two dimensional array to the output

    int n = 5;
    // nXn matrix
    int[][] myMatrix = new int[n][n];

    for (int row = 0; row < myMatrix.length; row++) {
      for (int column = 0; column < myMatrix[row].length; column++) {
        if (row == column) {
          myMatrix[row][column] = 1;
        } else {
          myMatrix[row][column] = 0;
        }
      }
      System.out.println(Arrays.toString(myMatrix[row]));
    }
    System.out.println("______________________________");
    System.out.println(Arrays.deepToString(myMatrix));
  }
}

File: python_solutions/week-02/day-1/arrays/diagonal_matrix.py
# - Create (dynamically) a two dimensional array
#   with the following matrix. Use a loop!
#
#   1 0 0 0
#   0 1 0 0
#   0 0 1 0
#   0 0 0 1
#
# - Print this two dimensional array to the output

diagonal_matrix = list()

for i in range(4):
    current_line = list()
    
    for j in range(4):
        current_line.append(1 if i == j else 0)
    else:
        diagonal_matrix.append(current_line)

for line in diagonal_matrix:
    print(line)
